src/routes/docs/tutorials/pagination/step-7/+page.markdoc

```diff
@@ -0,0 +1,265 @@
+    if (hasNext) {
+      setOffset(offset + pageLimit);
+    }
+  }
+
+  async function loadPrev() {
+    if (offset >= pageLimit) {
+      setOffset(offset - pageLimit);
+    }
+  }
+
+  async function loadPage(pageNumber) {
+    setOffset((pageNumber - 1) * pageLimit);
+  }
+
+  async function init() {
+    try {
+      const response = await databases.listDocuments(
+        TODOS_DATABASE_ID,
+        TODOS_COLLECTION_ID,
+        [
+          Query.orderAsc("$createdAt"),
+          Query.limit(pageLimit),
+          Query.offset(offset),
+        ],
+      );
+
+      setTodos(response.documents);
+      setCurrentPage(Math.floor(offset / pageLimit) + 1);
+      setCounts({
+        totalPages: response.total / pageLimit,
+        totalResults: response.total,
+      });
+    } catch (error) {
+      console.log(error);
+    }
+  }
+  useEffect(() => {
+    init();
+  }, [offset]);
+
+  return (
+    <TodosContext.Provider
+      value={{
+        current: todos,
+        loadNext,
+        loadPrev,
+        loadPage,
+        currentPage,
+        totalPages: counts.totalPages,
+        hasNext,
+        hasPrev: currentPage > 1,
+      }}>
+      {props.children}
+    </TodosContext.Provider>
+  );
+}
+```
+In this updated context:
+ * We're using `Query.limit()` to set the number of tasks we want to fetch per page.
+ * We're using `Query.offset()` to set the starting point for fetching tasks based on the current page.
+ * We've added `loadNextPage`, `loadPrevPage`, and `loadPage` function to handle pagination.
+
+ Now, you can use this context in your components to display todo items and handle pagination using Appwrite's offset pagination methods.
+
+# Update the Todos page {% #update-the-todoss-page %}
+
+Now, let's use our Pagination component in the Todos page.
+{% tabs %}
+{% tabsitem #Basic-pagination title="Basic Pagination" %}
+Update the Todos page in `src/pages/Todos.jsx`:
+```js
+import BasicPagination from '../components/BasicPagination';
+import {useTodos} from "../lib/context/todos";
+export function TodosWithBasicPagination() {
+  const todos = useTodos();
+
+  return (
+    <section>
+      <h2>Todos</h2>
+      <ul>
+        {todos.current.map(todo => (
+          <li key={`todo-${todo.taskId}`}>
+            <span>{`Task-${todo.taskId}: `}</span>
+            <input type='checkbox' id={todo.taskId} name={todo.title} />
+            <label htmlFor={todo.title}>{todo.title}</label>
+          </li>
+        ))}
+      </ul>
+      <BasicPagination
+        loadPrev={todos.loadPrev}
+        loadNext={todos.loadNext}
+        hasNext={todos.hasNext}
+        hasPrev={todos.currentPage > 1}
+      />
+    </section>
+  );
+}
+```
+{% /tabsitem %}
+{% tabsitem #numbered-pagination title="Numbered Pagination" %}
+Update the Todos page in `src/pages/Todos.jsx`:
+```js
+import NumberedPagination from '../components/NumberedPagination';
+
+export function TasksWithNumberedPagination() {
+  const tasks = useTasks();
```

Comment:
Where did this come from haha

Reply:
This would demo showing page jumping is possible with offset pagination but not cursor pagination.